__tests__/TestFuture.re

```diff
@@ -311,4 +311,38 @@ describe("Future Belt.Result", () => {
         |> finish
       );
   });
-});
+
+  testAsync("value recursion is stack safe", finish => {
+    let next = x => Future.value(~executor=`trampoline, x + 1);
+    let numberOfLoops = 10000;
+    let rec loop = x => {
+      next(x)
+      ->Future.flatMap(x' =>
+          if (x' == numberOfLoops) {
+            Future.value(x');
+          } else {
+            loop(x');
+          }
+        );
+    };
+    loop(0)
+    ->Future.get(r => r |> expect |> toEqual(numberOfLoops) |> finish);
+  });
+
+  testAsync("async recursion is stack safe", finish => {
```

Comment:
Make a test using the default executor to ensure it blows the stack and we're not just always using the trampoline

Reply:
See "value recursion blows the stack with default executor"